crates/sui-analytics-indexer/src/tables.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+#![allow(dead_code)]
+
+use serde::Serialize;
+use std::collections::BTreeSet;
+
+//
+// Table entries for the analytics database.
+// Each entry is a row in the database.
+//
+
+// Checkpoint information.
+#[derive(Serialize)]
+pub(crate) struct CheckpointEntry {
+    // indexes
+    pub(crate) checkpoint_digest: String,
+    pub(crate) sequence_number: u64,
+    pub(crate) epoch: u64,
+    pub(crate) timestamp_ms: u64,
+
+    pub(crate) previous_checkpoint_digest: Option<String>,
+    pub(crate) end_of_epoch: bool,
+    // gas stats
+    pub(crate) total_gas_cost: i64,
+    pub(crate) total_computation_cost: u64,
+    pub(crate) total_storage_cost: u64,
+    pub(crate) total_storage_rebate: u64,
+    // transaction stats
+    pub(crate) total_transaction_blocks: u64,
+    pub(crate) total_transactions: u64,
+    pub(crate) total_successful_transaction_blocks: u64,
+    pub(crate) total_successful_transactions: u64,
+
+    pub(crate) network_total_transaction: u64,
+    pub(crate) validator_signature: String,
+}
+
+// Transaction information.
+#[derive(Serialize)]
+pub(crate) struct TransactionEntry {
+    // main indexes
+    pub(crate) transaction_digest: String,
+    pub(crate) checkpoint: u64,
+    pub(crate) epoch: u64,
+    pub(crate) timestamp_ms: u64,
+    // transaction info
+    pub(crate) sender: String,
+    pub(crate) transaction_kind: String,
+    pub(crate) transaction_count: u64,
+    pub(crate) execution_success: bool,
+    // object info
+    pub(crate) input: u64,
+    pub(crate) shared_input: u64,
+    pub(crate) gas_coins: u64,
+    // objects are broken up in created, mutated and deleted.
+    // No wrap or unwrap information is provided
+    pub(crate) created: u64,
+    pub(crate) mutated: u64,
+    pub(crate) deleted: u64,
+    // PTB info
+    pub(crate) move_calls: u64,
+    pub(crate) packages: BTreeSet<String>,
```

Comment:
what does `packages` of a txn mean?

Reply:
in a sense I am not sure yet (lol), it is possibly a list of packages (comma separated or similar) used in the transaction.
The idea is to have an easy way via SQL (with regex match) to find all transactions that work against a certain package.
Thoughts?